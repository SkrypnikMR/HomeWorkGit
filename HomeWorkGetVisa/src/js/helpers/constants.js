export const namesArray = [
  `Максим`,
  `Кек`,
  `ДругКека`,
  `ПодругаКека`,
  `Лаврентий`,
  `Хесус`,
  `Импларио`,
  `Лена`,
  `Анатолий`,
  `Григорий`,
  `Крысарий`,
  `Саламейн`,
  `Систан`,
  `Слендермэн`,
  `СамоеПлохоеИмя`,
  `Галина`,
];

export const engLVLArray = ["A1", "A2", "B1", "B2", "C1", "C2"];
