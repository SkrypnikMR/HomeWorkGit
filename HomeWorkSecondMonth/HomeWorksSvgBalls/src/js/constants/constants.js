 export const MAX_WIDTH = 1400;
 export const MAX_HEIGHT = 700;
 export const MIN_RADIUS = 10;
 export const MAX_RADIUS = 35;
 export const MIN_SPEED = 10;
 export const MAX_SPEED = 30;
 export const TIMER = 25;
 export let allCircles = [];
