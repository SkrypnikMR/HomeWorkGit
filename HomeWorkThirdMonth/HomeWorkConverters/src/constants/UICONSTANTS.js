export const lengthOptions = [{
    id: 'Meters',
    value: 'Meters',
    name: 'Meters',
},
{
    id: 'Versts',
    value: 'Versts',
    name: 'Versts',
},
{
    id: 'Miles',
    value: 'Miles',
    name: 'Miles',
},
{
    id: 'Foots',
    value: 'Foots',
    name: 'Foots',
},
{
    id: 'Yards',
    value: 'Yards',
    name: 'Yards',
}];
export const currencyOptions = [{
    id: 'EUR',
    value: 'EUR',
    name: 'EUR',
},
{
    id: 'USD',
    value: 'USD',
    name: 'USD',
},
{
    id: 'RUR',
    value: 'RUR',
    name: 'RUR',
}];
export const currencyUAH = [{ id: 'UAH', value: 'UAH', name: 'UAH' }];
