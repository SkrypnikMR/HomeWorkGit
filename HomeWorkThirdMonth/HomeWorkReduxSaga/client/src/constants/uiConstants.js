export const defaultImage = 'assets/images/supportMovie.jpg';