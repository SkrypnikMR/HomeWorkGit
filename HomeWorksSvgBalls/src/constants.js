export const FULL_WIDTH = 1280;
export const FULL_HEIGHT = 720;
export const MAX_RADIUS = 20;
export const MIN_RADIUS = 5;
export const MAX_SPEED = 75;
export const MIN_SPEED = 25;
export const FRAME_INTERVAL = 20;
