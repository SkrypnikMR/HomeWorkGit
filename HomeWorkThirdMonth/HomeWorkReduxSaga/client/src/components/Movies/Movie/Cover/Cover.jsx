import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Input from '../../../Input';
import './Cover.scss';
import { NotificationManager } from 'react-notification';
import { defaultImage } from '../../../../constants/uiConstants';


const Cover = ({ name, id, cover, updateMovie }) => {
    const [state, setState] = useState({
        isWaitForChange: false,
        changeParam: 'cover',
        newData: '',
        error: false,
    });
    const handleOnChange = (e) => setState({ ...state, newData: e.target.value });
    const onError = () => setState({ ...state, error: true })
    const handleOnDoubleClick = () => setState((state) => ({ ...state, isWaitForChange: !state.isWaitForChange }));
    const handleOnBlur = async (e) => {
        if (e.target.value === '') {
            setState({ ...state, isWaitForChange: false });
            return NotificationManager.error('Please Enter newData for field', 'INPUT ERROR', 2000);
        }
        const { changeParam, newData } = state;
        setState({ ...state, isWaitForChange: false });
        await updateMovie({ changeParam, newData, id: id });
    }

    return (
        <>
            {
                state.isWaitForChange ? (
                    <div className="movie__cover">
                        <Input
                            name={name}
                            onBlurInput={handleOnBlur}
                            onChangeInput={handleOnChange}
                            placeholder='enter new cover'
                            value={state.newData}
                        />
                    </div>
                )
                    : <img src={state.error ? defaultImage : cover} onError={onError} onDoubleClick={handleOnDoubleClick} name={name} />
            }
        </>
    );

}

Cover.propTypes = {
    name: PropTypes.string.isRequired,
    cover: PropTypes.string,
    id: PropTypes.number.isRequired,
    updateMovie: PropTypes.func.isRequired
}

export default Cover;