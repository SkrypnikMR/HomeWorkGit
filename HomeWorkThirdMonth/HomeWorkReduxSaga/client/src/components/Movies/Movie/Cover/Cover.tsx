import React, { useState } from 'react';
import Input from '../../../Input';
import './Cover.scss';
import { NotificationManager } from 'react-notifications';
import { defaultImage } from '../../../../constants/uiConstants';

interface ICover {
    name: string;
    id: string;
    cover: string;
    updateMovie: (id: IpdateMovie) => void;
}

interface IpdateMovie {
    id: string;
    changeParam: string;
    newData: string;
}

const Cover = ({ name, id, cover, updateMovie }: ICover) => {
    const [state, setState] = useState({
        isWaitForChange: false,
        changeParam: 'cover',
        newData: '',
        error: false,
    });
    const handleOnChange = (e : React.ChangeEvent<HTMLInputElement>) => setState({ ...state, newData: e.target.value });
    const onError = () => setState({ ...state, error: true })
    const handleOnDoubleClick = () => setState((state) => ({ ...state, isWaitForChange: !state.isWaitForChange }));
    const handleOnBlur = async (e : React.ChangeEvent<HTMLInputElement>) => {
        if (e.target.value === '') {
            setState({ ...state, isWaitForChange: false });
            return NotificationManager.error('Please Enter newData for field', 'INPUT ERROR', 2000);
        }
        const { changeParam, newData } = state;
        setState({ ...state, isWaitForChange: false });
        await updateMovie({ changeParam, newData, id: id });
    }

    return (
        <>
            {
                state.isWaitForChange ? (
                    <div className="movie__cover">
                        <Input
                            name={name}
                            onBlurInput={handleOnBlur}
                            onChangeInput={handleOnChange}
                            placeholder='enter new cover'
                            value={state.newData}
                        />
                    </div>
                )
                    : <img src={state.error
                        ? defaultImage
                        : cover}
                        onError={onError}
                        onDoubleClick={handleOnDoubleClick}
                    />
            }
        </>
    );

}

export default Cover;