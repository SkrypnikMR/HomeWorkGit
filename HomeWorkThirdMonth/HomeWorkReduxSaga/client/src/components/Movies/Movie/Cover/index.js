import Cover from './Cover';
import { connect } from 'react-redux';
import { updateMovie } from '../../../../store/movies/actions';
import { getMovieField } from '../../../../store/movies/selectors';

const mapStateToProps = (state, props) => ({
    cover: getMovieField(state, props),
});

const mapDispatchToProps = { updateMovie };

export default connect(mapStateToProps, mapDispatchToProps)(Cover);