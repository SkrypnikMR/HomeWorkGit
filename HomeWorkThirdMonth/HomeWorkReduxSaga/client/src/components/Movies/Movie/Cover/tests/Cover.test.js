import React from 'react';
import { NotificationManager } from 'react-notifications';
import { shallow, mount } from 'enzyme';
import Cover from '../Cover';
import { act } from 'react-dom/test-utils';
import { defaultImage } from '../../../../../constants/uiConstants';


jest.mock('react-notifications', () => ({
    NotificationManager: {
        error: jest.fn()
    }
}))

describe('Cover', () => {
    let props;
    it('Should match snapshot', () => {
        props = {
            name: 'cover',
            id: 1,
            cover: 'someCover',
            updateMovie: jest.fn(),
        }
        const component = shallow(<Cover {...props} />);
        expect(component.html()).toMatchSnapshot();
    });
    it('Should render img', () => {
        props = {
            name: 'cover',
            id: 1,
            cover: 'someCover',
            updateMovie: jest.fn(),
        }
        const component = mount(<Cover {...props} />);
        expect(component.find('img')).toHaveLength(1);
    });
    it('Should render img with src === cover', () => {
        props = {
            name: 'cover',
            id: 1,
            cover: 'someCover',
            updateMovie: jest.fn(),
        }
        const component = mount(<Cover {...props} />);
        expect(component.find('img').props().src).toBe(props.cover);
    });
    it('Should render img with src === defaultImage', () => {
        props = {
            name: 'cover',
            id: 1,
            cover: 'someCover',
            updateMovie: jest.fn(),
        }
        const component = mount(<Cover {...props} />);
        act(() => component.find('img').props().onError());
        component.update();
        expect(component.find('img').props().src).toBe(defaultImage);
    });
    it('Should render Input ', () => {
        props = {
            name: 'cover',
            id: 1,
            cover: 'someCover',
            updateMovie: jest.fn(),
        }
        const component = mount(<Cover {...props} />);
        act(() => component.find('img').props().onDoubleClick());
        component.update();
        expect(component.find('img')).toHaveLength(0);
        expect(component.find('Input')).toHaveLength(1);
        expect(component.find('div').hasClass('movie__cover')).toBe(true);
    });
    it('Should change Input value ', () => {
        props = {
            name: 'cover',
            id: 1,
            cover: 'someCover',
            updateMovie: jest.fn(),
        }
        const testValue = 'testValue';
        const component = mount(<Cover {...props} />);
        act(() => component.find('img').props().onDoubleClick());
        component.update();
        act(() => component.find('Input').props().onChangeInput({ target: { value: testValue } }));
        component.update();
        expect(component.find('Input').props().value).toBe(testValue);
    });
    it('Should blur Input but with empty value ', async () => {
        props = {
            name: 'cover',
            id: 1,
            cover: 'someCover',
            updateMovie: jest.fn(),
        }
        const testValue = '';
        const component = mount(<Cover {...props} />);
        act(() => component.find('img').props().onDoubleClick());
        component.update();
        await act(async () => await component.find('Input').props().onBlurInput({ target: { value: testValue } }));
        component.update();
        expect(NotificationManager.error).toHaveBeenCalled();
    });
    it('Should blur Input without empty value ', async () => {
        props = {
            name: 'cover',
            id: 1,
            cover: 'someCover',
            updateMovie: jest.fn(),
        }
        const testValue = 'someValue';
        const component = mount(<Cover {...props} />);
        act(() => component.find('img').props().onDoubleClick());
        component.update();
        act(() => component.find('Input').props().onChangeInput({ target: { value: testValue } }));
        component.update();
        await act(async () => await component.find('Input').props().onBlurInput({ target: { value: testValue } }));
        component.update();
        expect(props.updateMovie).toHaveBeenCalledWith({ changeParam: props.name, id: props.id, newData: testValue })
    });
});
