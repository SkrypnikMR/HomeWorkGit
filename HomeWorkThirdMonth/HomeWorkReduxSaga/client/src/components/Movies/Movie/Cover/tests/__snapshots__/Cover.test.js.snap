// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cover Should match snapshot 1`] = `"<img src=\\"someCover\\" name=\\"cover\\"/>"`;
